src/plugins/navigation_embeddable/public/components/editor/navigation_embeddable_panel_editor_link.tsx

```diff
@@ -42,24 +45,85 @@ export const NavigationEmbeddablePanelEditorLink = ({
   parentDashboard?: DashboardContainer;
   dragHandleProps?: DraggableProvidedDragHandleProps;
 }) => {
+  const [dashboardError, setDashboardError] = useState<Error | undefined>();
   const parentDashboardTitle = parentDashboard?.select((state) => state.explicitInput.title);
   const parentDashboardId = parentDashboard?.select((state) => state.componentState.lastSavedId);
 
   const { value: linkLabel, loading: linkLabelLoading } = useAsync(async () => {
-    let label = link.label;
-    if (link.type === DASHBOARD_LINK_TYPE && !label) {
+    if (link.type === DASHBOARD_LINK_TYPE) {
       if (parentDashboardId === link.destination) {
-        label = parentDashboardTitle;
+        return link.label || parentDashboardTitle;
       } else {
-        const dashboard = await fetchDashboard(link.destination);
-        label = dashboard.attributes.title;
+        const dashboard = await fetchDashboard(link.destination)
+          .then((result) => {
+            setDashboardError(undefined);
+            return result;
+          })
+          .catch((error) => setDashboardError(error));
+        return (
+          link.label ||
```

Comment:
nit
```suggestion
          link.label ??
```

Reply:
Refer to https://github.com/elastic/kibana/pull/162285#discussion_r1287525000